uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -446,11 +440,7 @@ public function testSaveFromCalculationWillReplaceExistingLeavePeriodEntitlement
   }
 
   public function testSaveFromEntitlementCalculationCanSaveOverriddenValuesGreaterThanProposedEntitlement() {
-    // This mocks the logged in user so we can test
-    // the LeavePeriodEntitlement creation with a comment
-    global $user;
-    $user = new stdClass();
-    $user->uid = 1;
+    $this->registerCurrentLoggedInContactInSession(1);
```

Comment:
Took a look at why we actually need the `registerCurrentLoggedInContactInSession` function in this test, Its actually because of the comment. Also looking at the test title, it does not really need comments because it is testing another thing. I would suggest we remove registering and unregistering contact functions in this test and in the other test and pass comment as null to the `LeavePeriodEntitlement::saveFromCalculation` in both tests.
Then we need to add another test to verify that entitlements can  be saved properly with comments. In that test we will verify that the comment, comment_author_id and comment_date of the Entitlement are the correct values. Having this test will help prevent the bug you just fixed where the comment author was displaying wrongly.